src/vision_main.cpp

```diff
@@ -0,0 +1,127 @@
+#include "../include/vision.h"
+// #include "constants.h"
+#include <cstdlib>
+#include "opencv2/opencv.hpp"
+#include <stdio.h>
+#include <iostream>
+#include <jsoncpp/json/json.h>
+#include <memory>
+#include <fstream>
+
+std::vector<float> jsonToFloatVector(const Json::Value& jsonArray)
+{
+  std::vector<float> result;
+  if (!jsonArray.isArray())
+  {
+    std::cerr << "Input JSON is not an array!" << std::endl;
+    return result;
+  }
+  for (unsigned int i = 0; i < jsonArray.size(); i++)
+  {
+    if (jsonArray[i].isDouble() or jsonArray[i].isInt())
+    {
+      result.push_back(static_cast<float>(jsonArray[i].asFloat()));
+    }
+    else
+    {
+      std::cerr << "Element is not a float number!" << std::endl;
+    }
+  }
+  return result;
+}
+
+class CameraCalibration
```

Comment:
this probably belongs in vision.h. Should be part of the library and not the main executable, right?

Reply:
I was thinking on a dedicated file, so it enables people to use also yaml and other way to pass the information.

Reply:
sounds good